src/modules/validators/OwnableValidator.ts

```diff
@@ -1,179 +1,190 @@
+import { OWNABLE_VALIDATOR_ADDRESS, SENTINEL_ADDRESS, UserOperationStruct } from "../../account/index.js"
+import { BaseValidationModule } from "../base/BaseValidationModule.js"
+import type { Module } from "../utils/Types.js"
+
+export class OwnableValidator extends BaseValidationModule {
+    public smartAccount: NexusSmartAccount
+    public owners: Address[]
+    public threshold: number
+
+    private constructor(
+        moduleConfig: Module,
+        smartAccount: NexusSmartAccount
+    ) {
+        if (!moduleConfig.data) {
+            throw new Error("Module data is required")
+        }
+        const moduleData = decodeAbiParameters(
+            parseAbiParameters("uint256 threshold, address[] owners"),
+            moduleConfig.data
+        )
+        super(moduleConfig, smartAccount.getSmartAccountOwner())
+        this.threshold = Number(moduleData[0])
+        this.owners = [...moduleData[1]] as Address[]
+        this.smartAccount = smartAccount
+    }
+
+    public static async create(
+        smartAccount: NexusSmartAccount,
+        owners: Address[],
+        threshold?: number,
+        hook?: Address
+    ): Promise<OwnableValidator> {
+        if (
+            !owners.includes(await smartAccount.getSmartAccountOwner().getAddress())
+        ) {
+            throw Error("Signer needs to be one of the owners")
+        }
+        const installData = encodeAbiParameters(
+            [
+                { name: "threshold", type: "uint256" },
+                { name: "owners", type: "address[]" }
+            ],
+            [BigInt(threshold ?? owners.length), owners]
+        )
+        const moduleInfo: Module = {
+            moduleAddress: OWNABLE_VALIDATOR_ADDRESS,
+            type: "validator",
+            data: installData,
+            additionalContext: "0x",
+            hook
+        }
+        const instance = new OwnableValidator(moduleInfo, smartAccount)
+        return instance
+    }
+
+    /**
+     * Sets the threshold locally on the instance without executing a transaction.
+     * This method should be used when the threshold has been updated on-chain but the local instance hasn't been updated.
+     * 
+     * @param threshold The new threshold value
+     */
+    public setLocalThreshold(threshold: number): void {
+        if (threshold <= 0) {
+            throw new Error("Threshold must be a positive number")
+        }
+        this.threshold = threshold
+    }
+
+    public async setThresholdUserOp(threshold: number): Promise<Partial<UserOperationStruct>> {
+        const calldata = encodeFunctionData({
+            functionName: "setThreshold",
+            abi: parseAbi(["function setThreshold(uint256 _threshold) external"]),
+            args: [BigInt(threshold)]
+        })
+
+        const transaction = {
+            to: this.moduleAddress,
+            data: calldata,
+            value: 0n
+        }
+        const userOp = await this.smartAccount.buildUserOp([transaction]);
```

Comment:
I see benefit of passing smartAccount instance is userop can be sent right away. in stead of just requesting for calldata